src/backend/distributed/executor/executor.c

```diff
@@ -607,10 +679,33 @@ DistributedStatementRequiresRollback(DistributedPlan *distributedPlan)
 		return true;
 	}
 
+	/*
+	 * Checking the first task's placement list is not sufficient for all purposes since
+	 * for append/range distributed tables we might have unequal number of placements for
+	 * shards. However, it is safe to do here, because we're searching for a reference
+	 * table. All other cases return false for this purpose.
+	 */
 	task = (Task *) linitial(taskList);
 	if (list_length(task->taskPlacementList) > 1)
 	{
-		return true;
+		/*
+		 * Some tasks don't set replicationModel thus we only
+		 * rely on the anchorShardId, not replicationModel.
+		 *
+		 * TODO: Do we ever need replicationModel in the Task structure?
+		 * Can't we always rely on anchorShardId?
```

Comment:
can we resolve this todo?

Reply:
That's copied from `master`, so we can consider fixing it on [master](https://github.com/citusdata/citus/blob/674b7ce29a9822329a39367db7f1e68207ba745e/src/backend/distributed/executor/multi_router_executor.c#L817-L828).

I think that's also a candidate task for new joiner? I can open an issue for that?

Reply:
If that's not a blocker, I'd like to merge this PR @marcocitus

Reply:
ah, no worries then. :ship: